test/standalone/issue_13030/main.zig

```diff
@@ -0,0 +1,7 @@
+fn b(comptime T: type) ?@import("std").meta.FnPtr(fn () error{}!T) {
```

Comment:
Why is this not a behavior test? And skipping stage1 is preferable to using `std.meta.FnPtr`.

Reply:
In order to properly test the issue, I needed to avoid any other part of the program analyzing `std.builtin.StackTrace` for me.  Thich means I can't let any error-(union)-returning function get analyzed, so as a behavior test where everything gets analyzed together, there is no test failure even without the bugfix.  Additionally, the panic signature contains `std.builtin.StackTrace`, so even analyzing a default or custom panic handler would hide the issue, so I made it a `standalone` test to make sure it is tested with `build-obj`.

I just removed `std.mem.FnPtr` since an extra level of indirection doesn't affect the test anyway.